### PR TITLE
Ported Gitlab receive adapter to adapter/v2

### LINKED ISSUE #1200: Port Gitlab source to adapter/v2
**Problem**
Gitlab source still uses `knative.dev/eventing/pkg/adapter` module, which uses the cloudevents/sdk-go v1. We should port it to `knative.dev/eventing/pkg/adapter/v2`, which uses the new sdk-go

**[Persona:](https://github.com/knative/eventing/blob/master/docs/personas.md)**
Event developer/Event producer

**Exit Criteria**
Gitlab source should not use anymore `knative.dev/eventing/pkg/adapter`

**Time Estimate (optional):**
1

### PR DESCRIPTION
**Pull request by slinkydeveloper**
Signed-off-by: Francesco Guardiani <francescoguard@gmail.com>

Fixes #1200

### CODE REVIEW

File: gitlab/pkg/adapter/receive_adapter.go [hunk 1]
```diff
@@ -177,11 +176,16 @@ func (ra *gitLabReceiveAdapter) postMessage(payload interface{}, source, eventTy
 	event.SetID(eventID)
 	event.SetType(eventType)
 	event.SetSource(source)
-	event.SetDataContentType(cloudevents.ApplicationJSON)
-	event.SetData(payload)
+	err := event.SetData(cloudevents.ApplicationJSON, payload)
```
Comment (matzew) [inline gitlab/pkg/adapter/receive_adapter.go:179]: ah, that's actually nice

File: gitlab/pkg/adapter/receive_adapter.go [hunk 2]
```diff
@@ -177,11 +176,16 @@ func (ra *gitLabReceiveAdapter) postMessage(payload interface{}, source, eventTy
...
+	if err != nil {
+		return err
```
Comment (antoineco) [inline gitlab/pkg/adapter/receive_adapter.go:181]: Maybe return a better error context such as `"failed to set event data: %w", err`? From my experience the error message returned by `SetData` is too raw to be meaningful without context (like just "json error blah...")

**Comment by slinkydeveloper** [comment]
Looking in test-grid, the integration tests are broken on master: https://testgrid.knative.dev/eventing-contrib#nightly

**Pull request merged**